keps/sig-api-machinery/storage-migration.md

```diff
@@ -141,29 +149,57 @@ Read the [workflow section][] to understand how the API is used.
 ```golang
 type StorageVersionMigration struct {
   metav1.TypeMeta
-  // metadata.name should be "<resource>.<group>" of the resource being
-  // migrated.
-  // Because resouces can be served at different groups, in the future version,
-  // metadata.name can be hash(etcd instance, etcd path prefix).
+  // At the alpha stage, metadata.name must be "<resource>.<group>" of the
+  // resource being migrated.
+  // We require this name format to avoid duplicate `migration` CRs for the same
+  // resource type. Otherwise we will need to build complicated system to make
+  // sure the latest `migration` CR is honored. 
+  // Using "<resource>.<group>" isn't entirely safe, for two reasons:
+  // a. resources share key spaces can be served at different groups
+  // b. groupX/v1/foo and groupX/v5/foo may not share key spaces and thus
+  // require two `migration` CRs, though Kubernetes don't have such cases yet. 
+  // To address the two concerns, in the beta version, the name need to be
+  // tied to the key spaces, e.g., a hash of "etcd instance"+"etcd path prefix".
```

Comment:
The only reason I can think of for layering requirements on the name is for idempotency. Is that why you are doing it?

Why can't you just let the client worry about that? Why do you have to mandate it?